### PR TITLE
[core] Lay out casts and null checks more around gardening

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

### CODE REVIEW

File: src/map/utils/gardenutils.cpp [hunk 1]
```diff
@@ -95,15 +95,18 @@ namespace gardenutils
...
                         uint32 daysSinceStageChange = (vanatime - PPotItem->getStageTimestamp()) / VANADAY_SECONDS;
                         uint8  wiltTime             = VANADAYS_TO_WILT + PChar->getMod(Mod::GARDENING_WILT_BONUS);
                         bool   wasExamined          = PPotItem->wasExamined();
-                        if ((!wasExamined && (stageDuration > wiltTime || (stageDuration + daysSinceStageChange > wiltTime))) ||
-                            daysSinceStageChange > VANADAYS_TO_GUARANTEE_WILT + wiltTime)
+                        bool   ignoredDuringStage   = !wasExamined && stageDuration > wiltTime;
+                        bool   ignoredSinceChange   = !wasExamined && stageDuration + daysSinceStageChange > wiltTime;
+                        bool   shouldForceWilt      = daysSinceStageChange > VANADAYS_TO_GUARANTEE_WILT + wiltTime;
+
+                        if (ignoredDuringStage || ignoredSinceChange || shouldForceWilt)
```
Comment (zach2good) [inline src/map/utils/gardenutils.cpp:109]: I gave this logical split to downstream and they are currently testing it to see if one of these is redundant and possibly causing spurious wilting

**Comment by zach2good** [comment]
I'm going to be pretty busy this week, if someone could quickly smoke test this against a day or two of gardening it would be much appreciated

**Comment by Xaver-DaRed** [comment]
I made a fresh build of your branch.

Maybe I'm doing something wrong, however, after placing the flowerpots, when attempting to feed it a seed to grow it, it will just ignore the imput. Won't consume the seed and the flowerpot will remain empty.

This happens with all 10 placed flowerpots.

**Comment by zach2good** [comment]
I was hoping Horizon would have made some similar changes to this and had some feedback, but they haven't had time/bandwidth, and I don't know gardening well enough to weigh in on anything apart from how C++ works.

Closing, pending more reports/info

**Pull request closed without merging**